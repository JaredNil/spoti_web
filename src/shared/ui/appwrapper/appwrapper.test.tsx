import { render, screen } from '@testing-library/react'

import AppWrapper from '.'

describe('AppWrapper', () => {
	it('рендерит children', () => {
		render(
			<AppWrapper>
				<span data-testid="child">Hello</span>
			</AppWrapper>
		)
		expect(screen.getByTestId('child')).toBeInTheDocument()
	})

	it('имеет самые важные классы', () => {
		const { container } = render(<AppWrapper>test</AppWrapper>)
		const div = container.firstChild as HTMLElement

		expect(div).toHaveClass(
			'fixed',
			'top-0',
			'left-0',
			'right-0',
			'bottom-0'
		)
		expect(div).toHaveClass('w-dvw', 'h-vh')
		expect(div).toHaveClass('flex', 'flex-col')
		expect(div).toHaveClass('bg-black', 'scheme-dark', 'text-white')
		expect(div).toHaveClass('font-main')
	})

	it('snapshot', () => {
		const { container } = render(<AppWrapper>snap</AppWrapper>)
		expect(container.firstChild).toMatchSnapshot()
	})
})
