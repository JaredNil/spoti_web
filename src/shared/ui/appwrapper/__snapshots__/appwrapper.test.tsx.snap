// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AppWrapper snapshot 1`] = `
<div
  class="fixed top-0 left-0 right-0 bottom-0 w-dvw h-vh
            flex flex-col
            bg-black scheme-dark text-white
            font-main className"
>
  snap
</div>
`;
